src/libraries/System.Net.Sockets/tests/FunctionalTests/Connect.cs

```diff
@@ -82,6 +82,20 @@ public async Task Connect_MultipleIPAddresses_Success(IPAddress listenAt)
             }
         }
 
+        [OuterLoop("Slow on Windows")]
+        [Theory]
+        [InlineData(false)]
+        [InlineData(true)]
+        public virtual async Task Connect_WhenFails_ThrowsSocketExceptionWithIPEndpointInfo(bool useDns)
+        {
+            // Port 288 is unassigned:
+            // https://www.iana.org/assignments/service-names-port-numbers/service-names-port-numbers.txt
+            EndPoint badEndpoint = useDns ? (EndPoint)new DnsEndPoint("localhost", 288) : new IPEndPoint(IPAddress.Loopback, 288);
+            using Socket client = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp);
+            SocketException ex = await Assert.ThrowsAnyAsync<SocketException>(() => ConnectAsync(client, badEndpoint));
+            Assert.Contains("127.0.0.1:288", ex.Message);
```

Comment:
It may be safer to bind on anonymous port like the example above. I think that would fit better existing test pattern.